### ISSUE TITLE
[a11y]: DataTable [batch actions] - VO does not identify the beginning / ending of the Landmark region

### ISSUE THREAD

**Issue by andreancardona** [carbon-design-system/carbon#14258]
### Package

@carbon/react

### Browser

Safari & Iphone

### Package version

@carbon/react@1.32.0

### Automated testing tool and ruleset

Manual QA

### Assistive technology

VO 

### Description
Following landmark regions: data table toolbar region, Filter table search region

VO does not identify the beginning / ending of the Landmark region correctly when nested

VO does not announce the beginning of the "data table toolbar region". Similarly, it does not announce the ending of the "Filter table search region"

### Recommended Fix
This issue needs to be further investigated. We are not sure whether this is A Known VO bug. Will update this issue after cross checking with other working examples.

### Reproduction/example
Record `Number 61` in the below table:
https://airtable.com/shrFsaeHhImOkuVNR

https://react.carbondesignsystem.com/?path=/story/components-datatable-batch-actions--default

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems](https://github.com/carbon-design-system/carbon/issues/14255)

**Comment by francinelucca** [comment]
Landmark announcement: does not announce beginning of toolbar region and  it only announces beginning of search region but not end of search region. It also announces the ending of the toolbar region despite not announcing the beginning. Expected behavior is to announce both regions
